core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -197,7 +199,21 @@ private static Manifest loadLinkedManifest(File archive) throws IOException {
             if (libs != null)
                 paths.addAll(Arrays.asList(libs));
 
-            baseResourceURL = expandDir.toURI().toURL();
+            try {
+                Class pathJDK7 = Class.forName("java.nio.file.Path");
+                Object toPath = File.class.getMethod("toPath").invoke(expandDir);
+                URI uri = (URI) pathJDK7.getMethod("toUri").invoke(toPath);
+
+                baseResourceURL = uri.toURL();
+            } catch (NoSuchMethodException e) {
+                //Need to capture this exception because for JDK6 File.toPath() doesn't exist
```

Comment:
I think this can in fact `throw new Error(e)`, because if you are on Java 6 then you should have already gotten the `ClassNotFoundException` before you got to either `getMethod` call.